src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/FaultedTreeProjectItemContextProvider.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using Microsoft.VisualStudio.Shell;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Tree
+{
+    [Export(typeof(IProjectItemContextMenuProvider))]
+    [AppliesTo(ProjectCapability.CSharpOrVisualBasic)]
+    [Order(1)] // Ahead of CPS provider
+    internal class FaultedTreeProjectItemContextProvider : IProjectItemContextMenuProvider
```

Comment:
Let's get a bug filed against CPS to track adding the right context menu in this situation for post Dev15.